enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,297 @@
+---
+title: OVN-kubernetes live migration for hypershift kubevirt provider
+authors:
+  - @ellorent
+reviewers:
+  - "@dcbw"
+  - @danwinship
+  - @tssurya
+  - @numans
+approvers:
+  - TBD
+creation-date: 2023-02-06
+last-updated: 2023-02-06
+tracking-link: 
+  - "https://issues.redhat.com/browse/CNV-22946"
+see-also:
+  - "https://hypershift-docs.netlify.app/"
+---
+
+
+# OVN-kubernetes live migration for hypershift kubevirt provider
+
+## Summary
+
+The [Hypershift](https://hypershift-docs.netlify.app/) project creates openshift clusters on demand on top of already provisioned openshift by deploying  (and attaching working nodes to) "Hosted Control Planes" using one of different providers, such as aws, azure, baremetal (agent), and kubevirt.
+
+In KubeVirt's case, the worker nodes will be spun as KubeVirt virtual machines.
+
+There are some network requirements, some of them are imposed by kubevirt 
+live migration and others by kubernetes:
+
+- The IP Kubelet uses to register/communicate should follow the node during live migration.
+- Guest Cluster Network isolation. Guest clusters shouldn’t be able to talk to other guest clusters directly except through public LBs. Guest clusters shouldn’t be able to talk to any infra components except through public LBs.
+- No TCP connection breakage over E/W and N/S during live migration with minimal disruption.
+- Support exposing workers via services. 
+- Allow workers services access to services like kubedns and apiserver from hosted control plane.
+
+To accomplish the goals listed above the ovn-kubernetes default network has to implement some features triggered for an Hypershift KubeVirt VM.
+hypershift kubevirt VM is detected.
+
+## Motivation
+
+To reach GA at hypershift kubevirt provider it has to implement live migration and adding live-migration features to ovn-k default pod network
+is possible without much impact. This will allow users to spin hypershift kubevirt hosted clusters out of the box without setting up 
+the network at a secondary interface.
+
+### Goals
+
+Hypershift kubevirt provider needs the live-migration functionality, missing features at CNI need to be implement at the ovn-kuberentes 
+which is the default CNI for openshift.
+
+### Non-Goals
+
+- Support kubevirt live-migration out of hypershift context
```

Comment:
KubeVirt itself also requires this feature. 

Why is this a non-goal ? It makes it look like KubeVirt VMs do not require this. They do.

Reply:
Removed